src/darker/__main__.py

```diff
@@ -141,13 +141,23 @@ def modify_file(path: Path, new_content: str) -> None:
 def print_diff(path: Path, old_content: str, new_lines: List[str]) -> None:
     """Print ``black --diff`` style output for the changes"""
     relative_path = path.resolve().relative_to(Path.cwd()).as_posix()
-    difflines = list(
-        unified_diff(old_content.splitlines(), new_lines, relative_path, relative_path)
+    diff = "\n".join(
+        line.rstrip("\n")
+        for line in unified_diff(
+            old_content.splitlines(), new_lines, relative_path, relative_path,
+        )
     )
-    header1, header2, *rest = difflines
-    print(header1, end="")
-    print(header2, end="")
-    print("\n".join(rest))
+
+    if sys.stdout.isatty():
+        try:
+            from pygments import highlight
+            from pygments.formatters import TerminalFormatter
+            from pygments.lexers import DiffLexer
+
+            diff = highlight(diff, DiffLexer(), TerminalFormatter())
+        except ImportError:
+            pass
+    print(diff)
```

Comment:
Having `highlight()`, `DiffLexer()` and `TerminalFormatter()` inside the `try:` block theoretically could mask `ImportError`s happening inside them. Very probably would never happen, and even if it did, wouldn't matter at all. But as a general principle, isn't it good to limit code in the `try:` block to just those calls we want to catch errors from?

Here's one simplification which also avoids redefining the `diff` variable:
```suggestion
    if sys.stdout.isatty():
        try:
            from pygments import highlight
            from pygments.formatters import TerminalFormatter
            from pygments.lexers import DiffLexer
        except ImportError:
            print(diff)
        else:
            print(highlight(diff, DiffLexer(), TerminalFormatter()))
    else:
        print(diff)
```